draft-ietf-jsonpath-base.md

```diff
@@ -1485,87 +1494,44 @@ Notes:
 * `LogicalType` is an abstraction of the result of a `logical-expr`.
   Its two instances, `LogicalTrue` and `LogicalFalse`, are not related to
   the JSON literals `true` and `false` and have no syntactical representation in JSONPath.
+* `SingleNodeType` is an abstraction of a `singular-path` (which appears in a comparison,
+  in a test expression, or as a function argument).
 * `NodesType` is an abstraction of a `filter-path` (which appears
   in a test expression or as a function argument).
   Members of `NodesType` have no syntactical representation in JSONPath.
+* `SingleNodeType` is a subtype of `NodesType`. An instance of `SingleNodeType` is well-typed
+  wherever an instance of `NodesType` is well-typed.
```

Comment:
I don't like the "subtype" language.  It implies type inheritance, which is not a feature that our type system has.

Reply:
I used the term "subtype" as it's a pithy way of saying that `SingleNodeType` is acceptable everywhere `NodesType` is acceptable. For readers who think in terms of inheritance, a `SingleNodeType` _is a_ `NodesType` (i.e. any value of type `SingleNodeType` is also a value of `NodesType`).

What aspects of inheritance do you think are not applicable to the type system in this PR?

Reply:
I admit this is better than the usage than with the previous `Optional*` types.  Meh.  I'll let this go.

Reply:
> I used the term "subtype" as it's a pithy way of saying that `SingleNodeType` is acceptable everywhere `NodesType` is acceptable. 

I think I also would feel more comfortable if we said just that.

Reply:
Well, given subtyping concerns both of you, I've removed it in 6d968a4f.